.github/workflows/test-ubuntu-24.yml

```
@@ -0,0 +1,97 @@
+      if: always()
+      run: |
+        echo "Test Results Summary:"
+        echo "====================="
+        echo "Python Dependencies: ${{ steps.pip-install.outcome }}"
+        echo "Node Dependencies: ${{ steps.npm-install.outcome }}"
+        echo "Data Processing: ${{ steps.data-processing.outcome }}"
+        echo "Site Generation: ${{ steps.site-generation.outcome }}"
+    
+    - name: Upload Test Artifacts
+      if: always()
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)